apps/twap/src/app/routes/_chainId/_walletAddress/orders/index.ts

```diff
@@ -0,0 +1,185 @@
+  properties: {
+    chainId: {
+      enum: ['1', '5', '100'],
+    },
+    walletAddress: {
+      type: 'string',
+      minLength: ADDRESS_LENGTH,
+      maxLength: ADDRESS_LENGTH,
+    },
+  },
+} as const satisfies JSONSchema;
+type RouteSchema = FromSchema<typeof routeSchema>;
+
+const postOrderBodySchema = {
+  type: 'object',
+  required: ['order', 'safeTxHash'],
+  additionalProperties: false,
+  properties: {
+    order: {
+      type: 'object',
+      required: [
+        'sellToken',
+        'buyToken',
+        'receiver',
+        'partSellAmount',
+        'minPartLimit',
+        't0',
+        'n',
+        't',
+        'span',
+        'appData',
+      ],
+      additionalProperties: false,
+      properties: {
+        sellToken: {
+          type: 'string',
+          minLength: ADDRESS_LENGTH,
+          maxLength: ADDRESS_LENGTH,
+        },
+        buyToken: {
+          type: 'string',
+          minLength: ADDRESS_LENGTH,
+          maxLength: ADDRESS_LENGTH,
+        },
+        receiver: {
+          type: 'string',
+          minLength: ADDRESS_LENGTH,
+          maxLength: ADDRESS_LENGTH,
+        },
+        partSellAmount: {
+          type: 'string',
+        },
+        minPartLimit: {
+          type: 'string',
+        },
+        t0: {
+          type: 'number',
+        },
+        n: {
+          type: 'number',
+        },
+        t: {
+          type: 'number',
+        },
+        span: {
+          type: 'number',
+        },
+        appData: {
+          type: 'string',
+        },
+      },
+    },
+    safeTxHash: {
+      type: 'string',
+    },
+  },
+} as const satisfies JSONSchema;
+type PostOrderBodySchema = FromSchema<typeof postOrderBodySchema>;
+type PostOrderParamsSchema = RouteSchema;
+
+type GetOrdersParamsSchema = RouteSchema;
+
+export default async function (fastify: FastifyInstance) {
+  fastify.get<{
+    Params: GetOrdersParamsSchema;
+  }>(
+    '/',
+    {
+      schema: {
+        params: routeSchema,
+      },
+    },
+    async function (request, reply) {
+      const { chainId, walletAddress } = request.params;
+      const orderRepository = fastify.orm.getRepository(Order);
+
+      const orders = await orderRepository.find({
+        where: {
+          wallet: {
+            address: walletAddress,
+          },
+          chainId: Number(chainId),
+        },
+        relations: {
+          wallet: true,
+        },
+      });
+
+      reply.status(200).send(orders);
+    }
+  );
+
+  fastify.post<{ Body: PostOrderBodySchema; Params: PostOrderParamsSchema }>(
+    '/',
+    {
+      schema: {
+        params: routeSchema,
+        body: postOrderBodySchema,
+      },
+    },
+    async function (request, reply) {
+      const {
+        order: {
+          sellToken,
+          buyToken,
+          receiver,
+          partSellAmount,
+          minPartLimit,
+          t0,
+          n,
+          t,
+          span,
+          appData,
```

Comment:
Do you have to deconstruct all fields?
Would it be too bad to simply spread them onto the `create` method?

Reply:
Updated it.

Personally, it is more explicit - while we can write this with a spread, that tends to hide information which is why I don't prefer it.

Reply:
My personal preference is to pass along without deconstructing because:
1. One more place to (forget to) update when adding a new field to the API
2. If the API enforces no fields not defined in the schema, there shouldn't be anything weird coming in.
3. The schema is the single source of truth. Still additional transformations can occur here, but the bulk of the fields remain what the API expects